app/components/blacklight/gallery/slideshow_preview_component.rb

```diff
@@ -11,11 +11,23 @@ def initialize(document:, document_counter: nil, **args)
       end
 
       def before_render
-        thumbnail(presenter.thumbnail.render({ alt: presenter.heading })) if thumbnail.blank? && presenter.thumbnail.exists?
-        thumbnail(content_tag(:div, t('.missing_image', scope: [:blacklight_gallery]), class: 'thumbnail thumbnail-placeholder')) if thumbnail.blank?
+        set_default_thumbnail if thumbnail.blank?
         super
       end
 
+      def set_default_thumbnail
```

Comment:
I'm not sure I understand this method name or how it works. This is returning a value, but the value isn't used.  Can you add some explanation about that?

Reply:
This method populates the thumbnail slot with a value if one wasn't explicitly provided. Do you have a better name?

I'm also not sure what the concern about returning a value is? You'd like it to explicitly return nil, or...?

Reply:
How about `populate_thumbnail_slot`? I guess what is confusing here is that `#thumbnail` has the side effect of setting some stateful property, this is not a function. If you added a comment to the method (e.g. "populate the thumbnail slot with a value if one wasn't explicitly provided"), it would be immensely helpful to the reader.